### DISCUSSION TITLE
Pass full update object in all hooks !

### DISCUSSION THREAD

**Discussion by geminigeek** [payloadcms/payload#1007] [Feature Requests & Ideas]
hi,

i want to suggest an "originalUpdateObject" , similar to the data key which  is passed to beforeValidate hook and afterChange field hook , to be made available to all hooks down the line. this can be used to further process sub / nested objects  after  the main object is  updated or created.

currently we can access req.body while the update/create query is running with an express request, but while we are in local api , we don't have access to req.body?

mostly i will be using this new key in afterChange collection hook

this feature will be very helpful if we send a large object with many nested objects and arrays to save and update.

thanks for reading
Dee

**Comment by TimHal** [comment]
There is one PR open about passing the previous object state in the `afterChange` hook. 
https://github.com/payloadcms/payload/pull/1003

**Reply by geminigeek** [reply]
hi,

let me check this, but to simply put my feature request, i want to have access to the "data key which is passed to beforeValidate"  in all hooks down the line.
